### DISCUSSION TITLE
Bank Change midi message/automate loading new cartridges from DAW

### DISCUSSION THREAD

**Discussion by GodofPumpkins** [asb2m10/dexed#382] [General]
I'm doing a machine learning project using Dexed, and so to construct a dataset I'm trying to automate getting an audio sample from a large collection of patches. I can have my DAW, Reaper, send Dexed a note and then a patch change message, but that only gives me the sounds from a single cartridge. Is there any way to send a midi message or have a parameter for loading the next .syx file in a folder?

**Comment by probonopd** [comment]
I would imagine that we need to send a MIDI sysex message to the Dexed plugin.
Something roughly along these lines (I am not an expert in Reaper lua scripting myself, but this might get you pointed in the right direction):

```
-- Create new track
reaper.InsertTrackAtIndex(0, true)
local track = reaper.GetTrack(0, 0)

-- Add Dexed plugin to track
reaper.TrackFX_AddByName(track, "Dexed", false, 1)

-- Load the data from the syx
-- (to be implemented)
local sysex = ...

-- Create a media item at position 0
local mediaitem = reaper.CreateNewMIDIItemInProj(track, 0, 0)
local mediaitemtake = reaper.GetActiveTake(mediaitem)

-- Set length of media item to 1 bar
local length = 4 -- length in measures

-- Put the sysex into the track
reaper.MIDI_InsertTextSysexEvt(mediaitemtake, true, false, 0, -1, sysex)
```

When you then play the track, hopefully the sysex will be loaded into the Dexed plugin.
